tests/enterprise-contract/contract.go

```diff
@@ -51,23 +85,274 @@ var _ = framework.EnterpriseContractSuiteDescribe("Enterprise Contract E2E tests
+					// The BuildahDemo pipeline used to generate the test data does not
+					// include the required test tasks, so this policy should always fail.
+					defaultECP.Spec, ecp.SourceConfig{Include: []string{"test"}})
+				Expect(fwk.AsKubeAdmin.TektonController.CreateOrUpdatePolicyConfiguration(namespace, policy)).To(Succeed())
+				// printPolicyConfiguration(policy)
+
+				generator.Strict = true
+				pr, err := fwk.AsKubeAdmin.TektonController.RunPipeline(generator, namespace, pipelineRunTimeout)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fwk.AsKubeAdmin.TektonController.WatchPipelineRun(pr.Name, namespace, pipelineRunTimeout)).To(Succeed())
+
+				// Refresh our copy of the PipelineRun for latest results
+				pr, err = fwk.AsKubeAdmin.TektonController.GetPipelineRun(pr.Name, pr.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				tr, err := fwk.AsKubeAdmin.TektonController.GetTaskRunStatus(fwk.AsKubeAdmin.CommonController.KubeRest(), pr, "verify-enterprise-contract")
+				Expect(err).NotTo(HaveOccurred())
+
+				printTaskRunStatus(tr, namespace, *kubeClient.CommonController)
+				GinkgoWriter.Printf("Make sure TaskRun %s of PipelineRun %s failed\n", tr.PipelineTaskName, pr.Name)
+				Expect(tekton.DidTaskRunSucceed(tr)).To(BeFalse())
+				// Because the task fails, no results are created
+			})
+
+			It("fails when unexpected signature is used", func() {
+				secretName := fmt.Sprintf("dummy-public-key-%s", util.GenerateRandomString(10))
+				publicKey := []byte("-----BEGIN PUBLIC KEY-----\n" +
+					"MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAENZxkE/d0fKvJ51dXHQmxXaRMTtVz\n" +
+					"BQWcmJD/7pcMDEmBcmk8O1yUPIiFj5TMZqabjS9CQQN+jKHG+Bfi0BYlHg==\n" +
+					"-----END PUBLIC KEY-----")
+				GinkgoWriter.Println("Create an unexpected public signing key")
+				Expect(fwk.AsKubeAdmin.TektonController.CreateOrUpdateSigningSecret(publicKey, secretName, namespace)).To(Succeed())
+				generator.PublicKey = fmt.Sprintf("k8s://%s/%s", namespace, secretName)
+
+				pr, err := fwk.AsKubeAdmin.TektonController.RunPipeline(generator, namespace, pipelineRunTimeout)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fwk.AsKubeAdmin.TektonController.WatchPipelineRun(pr.Name, namespace, pipelineRunTimeout)).To(Succeed())
+
+				// Refresh our copy of the PipelineRun for latest results
+				pr, err = fwk.AsKubeAdmin.TektonController.GetPipelineRun(pr.Name, pr.Namespace)
+				Expect(err).NotTo(HaveOccurred())
+
+				tr, err := fwk.AsKubeAdmin.TektonController.GetTaskRunStatus(fwk.AsKubeAdmin.CommonController.KubeRest(), pr, "verify-enterprise-contract")
+				Expect(err).NotTo(HaveOccurred())
+
+				printTaskRunStatus(tr, namespace, *kubeClient.CommonController)
+				GinkgoWriter.Printf("Make sure TaskRun %s of PipelineRun %s failed\n", tr.PipelineTaskName, pr.Name)
+				Expect(tekton.DidTaskRunSucceed(tr)).To(BeFalse())
+				// Because the task fails, no results are created
+			})
+		})
+
+	})
+
+	Context("ec-cli command", func() {
```

Comment:
I think this is a bit of a misnomer since the tests here are actually testing the task, not the cli directly. Maybe these should go under the "verify-enterprise-contract task" context? Or maybe it just really needs a more accurate name.

Anyways, this is super minor. Feel free to ignore it :)

Reply:
Move both Context("ec-cli command") and Context("Release Policy") under the "verify-enterprise-contract task" context.